### PR TITLE
Simplified email registration

### PR DESCRIPTION
**Pull request by ice-myles**
Auth was changed from magic link to code. Removed status endpoint as obsolete. Separate endpoint for reset email change due to it requires to use jwt token payload at the email.

Requires to update translations.

### CODE REVIEW

File: cmd/eskimo-hut/api/docs.go [hunk 1]
```diff
@@ -20,12 +20,9 @@ const docTemplate = `{
     "host": "{{.Host}}",
     "basePath": "{{.BasePath}}",
     "paths": {
-        "/auth/getConfirmationStatus": {
+        "/auth/ResetEmailChange": {
```
Comment (ice-ares) [inline cmd/eskimo-hut/api/docs.go]: Start with lowercase r

File: cmd/eskimo-hut/auth.go [hunk 2]
```diff
@@ -98,12 +154,14 @@ func (s *service) SendSignInLinkToEmail( //nolint:gocritic,funlen // .
 //	@Failure		422		{object}	server.ErrorResponse	"if syntax fails"
 //	@Failure		500		{object}	server.ErrorResponse
 //	@Failure		504		{object}	server.ErrorResponse	"if request times out"
-//	@Router			/auth/signInWithEmailLink [POST].
-func (s *service) SignIn( //nolint:gocritic //.
+//	@Router			/auth/ResetEmailChange [POST].
```
Comment (ice-ares) [inline cmd/eskimo-hut/auth.go]: lowercase r

File: cmd/eskimo-hut/auth.go [hunk 3]
```diff
@@ -98,12 +154,14 @@ func (s *service) SendSignInLinkToEmail( //nolint:gocritic,funlen // .
 //	@Failure		422		{object}	server.ErrorResponse	"if syntax fails"
 //	@Failure		500		{object}	server.ErrorResponse
 //	@Failure		504		{object}	server.ErrorResponse	"if request times out"
-//	@Router			/auth/signInWithEmailLink [POST].
-func (s *service) SignIn( //nolint:gocritic //.
+//	@Router			/auth/resetEmailChange [POST].
+//
+//nolint:gocritic,dupl //.
+func (s *service) ResetEmailChange(
```
Comment (ice-cronus) [inline cmd/eskimo-hut/auth.go]: we'll need not to forget to notify web-app devs as well that they shoud submit that confirmation code and payload to another endpoint from that code-input web-app page (if it still exists).

File: auth/email_link/DDL.sql [hunk 4]
```diff
@@ -9,7 +9,7 @@ CREATE TABLE IF NOT EXISTS email_link_sign_ins (
            previously_issued_token_seq            BIGINT DEFAULT 0 NOT NULL,
            confirmation_code_wrong_attempts_count BIGINT DEFAULT 0 NOT NULL,
            email                                  TEXT NOT NULL,
-           otp                                    TEXT NOT NULL,
+           otp                                    TEXT,
```
Comment (ice-cronus) [inline auth/email_link/DDL.sql]: I am not sure if we need that otp and email payloads at all anymore, cuz it was to check if user clicked right link / email (and confirmation code was 2nd factor). And for now we identify it only by code, so otp and magicLink token seems to be redundant. But we can keep at as is as it requires less modifications (including web-app side)

**Review by ice-cronus — DISMISSED** [review]

File: auth/email_link/link_verify.go [hunk 5]
```diff
@@ -17,56 +17,74 @@ import (
...
+		return nil, false, errors.Wrapf(err, "failed to get user info by email:%v(old email:%v)", id.Email, token.OldEmail)
 	}
-	if vErr := c.verifySignIn(ctx, els, &id, emailLinkPayload, confirmationCode, token.OTP); vErr != nil {
-		return errors.Wrapf(vErr, "can't verify sign in for id:%#v", id)
+	emailConfirmed, err = c.signIn(ctx, now, els, &id, token.OldEmail, token.NotifyEmail, confirmationCode)
+	if err != nil {
+		return nil, false, errors.Wrapf(err, "can't sign in for email:%v, deviceUniqueID:%v", id.Email, id.DeviceUniqueID)
+	}
+	els.TokenIssuedAt = now
+	tokens, err = c.generateTokens(els.TokenIssuedAt, els, els.IssuedTokenSeq+1)
```
Comment (ice-cronus) [inline auth/email_link/link_verify.go]: this approach is less atomic, you need returning most probably

**Review by ice-cronus — CHANGES_REQUESTED** [review]

**Review by ice-cronus — APPROVED** [review]

**Pull request merged**